sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -495,3 +495,20 @@ def test_S_srepr_is_identity():
     p = Piecewise((10, Eq(x, 0)), (12, True))
     q = S(srepr(p))
     assert p == q
+
+
+def test_issue_10258():
+    assert Piecewise((0, x < 1), (1, True)).is_zero is None
+    assert Piecewise((-1, x < 1), (1, True)).is_zero is False
+    a = Symbol('a', zero=True)
+    assert Piecewise((0, x < 1), (a, True)).is_zero
+    assert Piecewise((1, x < 1), (a, x < 3)).is_zero is None
+    a = Symbol('a')
+    assert Piecewise((0, x < 1), (a, True)).is_zero is None
+    assert Piecewise((0, x < 1), (1, True)).is_nonzero is None
+    assert Piecewise((1, x < 1), (2, True)).is_nonzero
+    assert Piecewise((0, x < 1), (oo, True)).is_finite is None
+    assert Piecewise((0, x < 1), (1, True)).is_finite
+    assert Piecewise(([1], x < 1)).is_finite is None
```

Comment:
I was surprised, too, but this is necessary as long as anything in SymPy can emit a list, e.g. solve. A Piecewise is an appropriate container for handling any objects (Basic or otherwise) whose selection depends on some condition. The only way to represent a solution to an equation with piecewise conditions is with a Piecewise of lists (or dicts).

For a univariate Piecewise we can give this:

```
>> solve(Piecewise((x**2-4,x>-4),(x+6,True)), x)
[-6, -2, 2]
```
But for a multivariate one we need to give a piecewise result:
```
>> solve(Piecewise((x**2-4,y>0),(x+6,True)),x)
Piecewise(([-2, 2], y > 0), (-6, True))
```